pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoParserTools.java

```diff
@@ -216,6 +217,9 @@ public static double readDouble(final ReadableSequentialData input) throws IOExc
      */
     public static String readString(final ReadableSequentialData input) throws IOException {
         final int length = input.readVarInt(false);
+        if (input.remaining() < length) {
+            throw new BufferOverflowException();
+        }
         byte[] bytes = new byte[length];
         input.readBytes(bytes);
```

Comment:
I think we need to check number of bytes read from readBytes as well. `(input.remaining() < length)` might well be true because we are reading from a stream where remaining is infinite but we still did not read all bytes. We can never safely ignore number of bytes read when using streams.